app/src/main/java/com/android/ubclaunchpad/driver/session/SessionInfoActivity.java

```diff
@@ -215,14 +180,148 @@ public void onCancelled(DatabaseError databaseError) {
                 });
     }
 
+    private void initializeListeners(final ArrayAdapter<String> adapter, final String sessionName) {
```

Comment:
this is a really big Activity since you're making 3 listeners...maybe you could move some of the code into a new non-Activity class?